sql/catalyst/src/main/scala/org/apache/spark/sql/types/UDTRegistration.scala

```diff
@@ -29,7 +29,6 @@ import org.apache.spark.util.Utils
  * However, by doing this, we add SparkSQL dependency on user classes. This object provides
  * alternative approach to register UDTs for user classes.
  */
-private[spark]
```

Comment:
Alternatively, we can add an interface to `UDTRegistration`, instead of making it public.